### PR TITLE
ipatests: test Samba mount with NTLM authentication

### PR DESCRIPTION
**Pull request by wladich**
Related to https://pagure.io/freeipa/issue/8636

### CODE REVIEW

**Comment by abbra** [comment]
Test looks good but it needs to be wrapped with FIPS check because NTLMSSP will not work in FIPS mode. Otherwise good.

**Comment by wladich** [comment]
Yes, I agree, I am just waiting for test run with FIPS to see it failing.

**Comment by wladich** [comment]
Test were modified:
* added test cases for specifying domain name as a part of UPN
* decorator for skipping tests in FIPS mode is moved to `patests.pytest_ipa.integration` package

**Comment by wladich** [comment]
Succesful test run for test_smb: [report](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/30e25b36-72c4-11eb-97a0-fa163e607052/report.html)

Removing temp commit

File: ipatests/pytest_ipa/integration/__init__.py [hunk 1]
```diff
@@ -478,3 +480,18 @@ def del_compat_attrs(cls):
         del cls.ad_subdomains
         del cls.ad_treedomains
     del cls.ad_domains
+
+
+def skip_if_fips(reason='Not supported in FIPS mode', host='master'):
```
Comment (kaleemsiddiqu) [inline ipatests/pytest_ipa/integration/__init__.py:485]: @wladich 
Why we are not using pyest skipif decorator (https://docs.pytest.org/en/reorganize-docs/new-docs/user/skipping.html) with call for fips.is_fips_enabled? 
May be there could be scenario where we need to skip tests in FIPS mode due to some bug, so in that case it won't fit.

Comment (wladich) [reply ipatests/pytest_ipa/integration/__init__.py:485]: @kaleemsiddiqu 

* `skipif` requires a boolean (not a callable) as as condition. 
* `is_fips_enabled` requires `host` argument, an instance of Host
*  the decorator is executed when the test class (not the instance) is created. 
*  Host objects are created in `mh` fixture, which is called for the test class instance 

=> When decorator is executed the Host object for test machine does not yet exist. 


I do not fully get the second part of the comment. We actually have a bug here and we skip the test until it is fixed.

Comment (kaleemsiddiqu) [reply ipatests/pytest_ipa/integration/__init__.py:485]: Second part is about specific reason displayed e.g skipping due to a bz which i think can be used in the current implementation.

**Review by amore17 — CHANGES_REQUESTED** [review]
test_file_client_path also needs to be removed.

**Comment by wladich** [comment]
> test_file_client_path also needs to be removed.

No. `test_file_client_path` is a view of remote file on client side inside the mounted directory.

**Comment by abbra** [comment]
ACK.

**Comment by wladich** [comment]
master:

* 0b60408dab4aee3a02448d0147f35433da446c61 ipatests: test Samba mount with NTLM authentication

**Comment by wladich** [comment]
Not backporting test to ipa-4-6 as it is missing `ipa-client-samba` utility whichis required by `test_smb` test suite.

**Pull request closed without merging**